R/calculate_transition_disruption_metric.R

```diff
@@ -0,0 +1,121 @@
+#' Calculate the transition disruption metric, based on the IPR scenario
+#'
+#' @param data A dataset like the `*_results_portfolio.rda` outputs of PACTA.
+#' @param start_year The start year for which the TDM value should be calculated.
+#' @param ... Variables to group by.
+#'
+#' @return A tibble with the columns specified in the `...` input, as well as:
+#'   `tdm_tech`, the technology level transition disruption metric and `tdm_sec`,
+#'    the sector level transition disruption metric.
+#' @export
+#'
+#' @examples
+calculate_transition_disruption_metric <- function(data, start_year, ...) {
+  stopifnot(
+    is.data.frame(data),
+    is.numeric(start_year)
+  )
+
+  crucial <- c(
+    "allocation",
+    "ald_sector",
+    "technology",
+    "year",
+    "scen_alloc_wt_tech_prod",
+    "plan_alloc_wt_tech_prod",
+    "plan_carsten"
+  )
+
+  check_crucial_names(data, crucial)
+
+  groups <- c(
+    "technology",
+    "ald_sector",
+    ...
+  )
```

Comment:
Note to self: `...` seems to expect characters. Test behavior with symbols (as in `dplyr::group_by()`) and compare with existing functions that take grouping variables via `...`, e.g. `dplyr::count()`. I guess we want to be consistent.

Reply:
For now I followed the example at the top of this PR and removed `...` in favor of a character vector. This seems to do the job. We may change this later for a more natural feel. But for now let's focus on behaviour.